ADR/0032-decoupling-deployment.md

```diff
@@ -0,0 +1,201 @@
+
+**During onboarding**: whereas today when a user requests a new appstudio tier namespace, the tier
+template includes an [Environment] that the integration-service will promote to. Tomorrow, the
+appstudio tier template should no longer include an [Environment] which on its own will cause
+integration-service to _not_ trigger a deployment when testing completes. Instead, the appstudio
+tier template should include a [ReleasePlan] with a reference to the [push-to-registry]
+release pipeline. This new default [ReleasePlan] should carry parameters such that whenever
+a [Snapshot] is successfully tested, a [Release] is created that re-tags the images in their same
+quay repositories with a tag like `:released` or `:validated` (name tbd). The
+[push-to-registry] pipeline can use the `appstudio-pipeline` service account in the user's
+dev workspace, which already has push access to the repository in question.
+
+```mermaid
+flowchart TD
+    NSTierTemplate --> |provides| ReleasePlan
+    A[fa:fa-code-commit Git Commit] --> |webhook| PipelinesAsCode
+    PipelinesAsCode --> |creates| build[Build PipelineRun]
+    build --> |triggers| integration-service
+    integration-service --> |asks| decision{Is there a ReleasePlan?}
+    ReleasePlan --> |influences| decision
+    decision --> |yes| Release
+    build --> |pushes the image| quay.io/redhat-user-workloads/tenant/component
+    Release --> |tags the image for release| quay.io/redhat-user-workloads/tenant/component
+    quay.io/redhat-user-workloads/tenant/component --> renovatebot[User provided RenovateBot]
+    renovatebot --> |updates| gitops-repo[User provided GitOps repo]
+    gitops-repo --> |causes deployment via| argocd[User provided ArgoCD]
+    argocd --> |deploys to| cluster[User provided cluster]
+```
+
+Outside of the AppStudio member cluster, the user is responsible for acquiring a gitops repo and
+deployment environments of their choice, manually laying out their application resources in the repo
+(assisted by tools like `kam`), specifying image references by tag to match the `:released` or
+`:validated` tagging schem mentioned above, configuring Argo to deploy from their gitops repo, and
+configuring [renovatebot] to propagate image updates by digest to their gitops repo.
+
+**For manual creation of new environments** - the user manages this directly using a combination of
+their gitops repo and argo, outside of the AppStudio member cluster.
+
+**For automated testing in ephemeral environments** - the user specifies an
+[IntegrationTestScenario] CR with a [DeploymentTargetClass] reference. After a build completes, but
+before it executes tests, the integration-service creates a new [DeploymentTargetClaim] CR with
+a [DTCLS] matching the one on the [IntegrationTestScenario]. This should cause the
+[deployment-target-operator] to provision and bind a [DeploymentTarget] matching the
+[DeploymentTargetClass]. [integration-service] then passes the kubeconfig and [Snapshot] to the
```

Comment:
Nit: 
```suggestion
[DeploymentTargetClass]. [Integration-service] then passes the kubeconfig and [Snapshot] to the
```
I _assume_ that this would invalidate the common link shortcut using a lowercase `i`.